### ISSUE TITLE
Auto orientation problem

### ISSUE THREAD

**Issue by alchemydigital** [thephpleague/glide#47]
Not strictly a bug with Glide, but more a limitation of Intervention that affects Glide functionality.  As mentioned in [this issue](https://github.com/Intervention/image/issues/336) Intervention can only read EXIF data if the image object is initialised with a file path (since it uses `exif_read_data`).  Since Glide always passes in the file contents (at it is loading the image via Flysystem) this means that auto-orientation will never work as there will be no EXIF information for Intervention to use.  Explicit orientation (i.e. rotate 90deg) is unaffected.

A quick and dirty fix that we have employed is to update the `makeImage()` method in `League\Glide\Server\Server.php` to write the source file to a temporary file in the system `tmp/` folder and then pass that path to the API run method (with appropriate garbage collection to clear out the temporary file once it has been used):

```php
$source = $this->source->read(
    $this->getSourcePath($request)
);

if ($source === false) {
    throw new FilesystemException(
        'Could not read the image `'.$this->getSourcePath($request).'`.'
    );
}

// Write the source to a temporary file
$tmp = tempnam(sys_get_temp_dir(), '');
$handle = fopen($tmp, "w");
fwrite($handle, $source);

try {
    $write = $this->cache->write(
        $this->getCachePath($request),
        // Pass the path to the temporary file instead of the source itself
        $this->api->run($request, $tmp)
    );
} catch (FileExistsException $exception) {
    // Cache file failed to write. Fail silently.
    // Remove the temporary file
    unlink($tmp);
    return $request;
}

// Remove the temporary file
unlink($tmp);
```

Happy to make a PR with this suggested change - even happier to see a more elegant solution!

**Comment by reinink** [comment]
Hmm, that is NOT good. Thank you very much for finding this—that makes total sense. Let me think about how best to solve this...your temp file solution may be the only good option we have, but let me do some digging first.

**Comment by renege** [comment]
Update?

**Comment by reinink** [comment]
@renege Soon. Very sorry for the delay. See [this](https://twitter.com/reinink/status/590132420558639104) and maybe [this](https://twitter.com/reinink/status/590132439986610176).

**Comment by alchemydigital** [comment]
It's also simple enough to do something like [this](https://gist.github.com/alchemydigital/2e3a2b7261e36003e42f) and [this](https://gist.github.com/alchemydigital/ab0eb238910e9c8a18c4) whilst @reinink works through his backlog.

**Comment by soee** [comment]
+1 for some fix for this

**Comment by reinink** [comment]
This has been fixed (57a10a4b8049f9efb7debb5cfd9718f897e205d3) and tagged in 0.3.5.
